superset/assets/src/chart/chartAction.js

```diff
@@ -186,3 +186,23 @@ export function runQuery(formData, force = false, timeout = 60, key) {
     ]);
   };
 }
+
+export function redirectSQLLab(formData) {
+  return function () {
+    $.ajax({
+      type: 'GET',
+      url: `/superset/sql?form_data=${JSON.stringify(formData)}`,
+      success: (response) => {
+        const url = new URL(window.location);
+        url.pathname = '/superset/sqllab';
+        for (let k of url.searchParams.keys()) {
+          url.searchParams.delete(k);
+        }
+        url.searchParams.set('datasourceKey', formData.datasource);
+        url.searchParams.set('sql', response.query);
+        window.open(url.href, '_blank');
```

Comment:
@mistercrunch, is there a better way of doing this? I wasn't sure how to get the base URL from JS.

Reply:
I think it's a bit the wild west in terms of URL composition in Superset at the moment. It could be good to refactor a `src/utils/urls.js` module with constants and utilities for all this.